Add notes and questions for #24571 (#491)
bitcoin-core-review-club/website: _posts/2022-04-06-#24571.md

commit:
---

## Notes

> * Fingerprinting allows an attacker to recognize the same node across different

Reviewer: ```suggestion
* Attackers may use fingerprinting techniques to recognize the same node across different
```

Reviewer: Could also link to a wiki article since it's a broad term https://en.wikipedia.org/wiki/Device_fingerprint

  connections. This makes it possible to test if two addresses belong to the
  same node, which we generally try to avoid especially for addresses belonging
  to privacy centric networks such as Tor. Some fingerprinting attacks work
  across restarts of the victims node, making it possible to detect if a node
  changes addresses.

* A variety of fingerprinting techniques have been previously patched. Some
  examples include: [#9268](https://github.com/bitcoin/bitcoin/pull/9268),
  [#8408](https://github.com/bitcoin/bitcoin/pull/8408),
  [#23695](https://github.com/bitcoin/bitcoin/pull/23695),
  [#11113](https://github.com/bitcoin/bitcoin/pull/11113),
  [#11740](https://github.com/bitcoin/bitcoin/pull/11740),
  [#5860](https://github.com/bitcoin/bitcoin/pull/5860),

>   [#5820](https://github.com/bitcoin/bitcoin/pull/5820). As shown by these PRs,

Reviewer: To make the notes more detailed, could give summaries when you mention each PR, maybe just one sentence or a few words? E.g. "attackers may use [addr message timestamps](), [pruning depth](), and [responses to p2p announcements]() to fingerprint nodes"

  fingerprinting can be found in all sorts of different components such as the
  address manager, block index or pruning logic.

> 

Reviewer: Missing a little more background on  PR #24571? Maybe a quick summary of what the fingerprintable behavior is?
